### PR TITLE
[cli] pulsar-client-tools supports end-to-end encryption

### PR DESCRIPTION
**Pull request by massakam**
I enabled the `pulsar-client` command to encrypt message payloads.

How to use:
```sh
$ ./bin/pulsar-client produce \
  -m my-msg \
  -n 10 \
  -ekn my-app-key \
  -ekv file:///path/to/public.key \
  persistent://public/default/test

$ ./bin/pulsar-client consume \
  -s my-sub \
  -n 10 \
  -ekv file:///path/to/private.key \
  persistent://public/default/test
```

### CODE REVIEW

File: pulsar-client-tools-test/src/test/java/org/apache/pulsar/client/cli/PulsarClientToolTest.java [hunk 1]
```diff
@@ -209,4 +210,55 @@ public void testDurableSubscribe() throws Exception {
...
+        final String topicName = "persistent://prop/ns-abc/test/topic-" + UUID.randomUUID().toString();
+        final int numberOfMessages = 10;
+
+        ExecutorService executor = Executors.newSingleThreadExecutor();
+        CompletableFuture<Void> future = new CompletableFuture<>();
+        executor.execute(() -> {
+            try {
+                PulsarClientTool pulsarClientToolConsumer = new PulsarClientTool(properties);
+                String[] args = {"consume", "-s", "sub-name", "-n", Integer.toString(numberOfMessages), "-ekv",
+                        "file:./src/test/resources/crypto_rsa_private.key", topicName};
```
Comment (tuteng) [inline pulsar-client-tools-test/src/test/java/org/apache/pulsar/client/cli/PulsarClientToolTest.java:229]: Consider reusing the files `private-key.client-rsa.pem` and `public-key.client-rsa.pem` under this folder `https://github.com/apache/pulsar/tree/master/pulsar-broker/src/test/resources/certificate` to test end-to-end encryption

File: pulsar-client-tools-test/src/test/java/org/apache/pulsar/client/cli/PulsarClientToolTest.java [hunk 2]
```diff
@@ -209,4 +210,55 @@ public void testDurableSubscribe() throws Exception {
...
+                    break;
+                }
+            } catch (Exception e) {
+            }
+            Thread.sleep(200);
+        }
+
+        PulsarClientTool pulsarClientToolProducer = new PulsarClientTool(properties);
+        String[] args = {"produce", "-m", "Have a nice day", "-n", Integer.toString(numberOfMessages), "-ekn",
+                "my-app-key", "-ekv", "file:./src/test/resources/crypto_rsa_public.key", topicName};
```
Comment (tuteng) [inline pulsar-client-tools-test/src/test/java/org/apache/pulsar/client/cli/PulsarClientToolTest.java:251]: Consider reusing the files `private-key.client-rsa.pem` and `public-key.client-rsa.pem` under this folder `https://github.com/apache/pulsar/tree/master/pulsar-broker/src/test/resources/certificate` to test end-to-end encryption

File: pulsar-client-tools-test/src/test/java/org/apache/pulsar/client/cli/PulsarClientToolTest.java [hunk 3]
```diff
@@ -209,4 +210,55 @@ public void testDurableSubscribe() throws Exception {
...
+                        "file:./src/test/resources/crypto_rsa_private.key", topicName};
+                Assert.assertEquals(pulsarClientToolConsumer.run(args), 0);
+                future.complete(null);
+            } catch (Throwable t) {
+                future.completeExceptionally(t);
+            }
+        });
+
+        // Make sure subscription has been created
+        while (true) {
```
Comment (eolivelli) [inline pulsar-client-tools-test/src/test/java/org/apache/pulsar/client/cli/PulsarClientToolTest.java:238]: can you use Awaitility ?

**Review by eolivelli — CHANGES_REQUESTED** [review]
great! very useful.
I left a little comment about tests

Do we need the same fix for `pulsar-perf` client ?

**Comment by massakam** [comment]
Addressed the comments. PTAL

**Review by merlimat — APPROVED** [review]

**Pull request merged**